### ISSUE TITLE
[Bug] Event was skipped when indexed by subgraph

### ISSUE THREAD

**Issue by PushinSt** [graphprotocol/graph-node#4647]
### Bug report

Hello. I found a problem that I can not find an entry in the subgraph about a specific transaction. Could it be that the block was not indexed?  

Graph:  
https://thegraph.com/hosted-service/subgraph/uniswap/uniswap-v3  

Transaction:  
https://etherscan.io/tx/0xfdeca9f81d0f930e7e107f10b26fd79a26e67b7edf41c683c4d00038e7ccd6f2#eventlog  

Event:
Burn  

Request for the transaction that was not indexed:
```
query txnBurn_failed {
  transactions(
    where: {id: "0xfdeca9f81d0f930e7e107f10b26fd79a26e67b7edf41c683c4d00038e7ccd6f2"}
  ) {
    id
    burns {
      id
      pool {
        id
      }
    }
  }
}
```

An example of a similar transaction that has been indexed
```
query txnBurn_example {
  transactions(
    where: {id: "0x000319c5821b918eecea3863f597ca5479c95970b3998588583ad2f6133b4ba1"}
  ) {
    id
    burns {
      id
      pool {
        id
      }
    }
  }
}
```

### Subgraph name or link to explorer

https://thegraph.com/hosted-service/subgraph/uniswap/uniswap-v3

### Some information to help us out

- [ ] Tick this box if this bug is caused by a regression found in the latest release.
- [ ] Tick this box if this bug is specific to the hosted service.
- [X] I have searched the issue tracker to make sure this issue is not a duplicate.

### OS information

None

**Comment by incrypto32** [comment]
My guess, would be the RPC provider itself missed the event at the time of processing, the `graph-node` relies on the RPC for event logs. And we've seen RPC's returning wrong logs occasionally in the past.

**Comment by PushinSt** [comment]
> My guess, would be the RPC provider itself missed the event at the time of processing, the `graph-node` relies on the RPC for event logs. And we've seen RPC's returning wrong logs occasionally in the past.

Wow! but how to fix it? I have a function that calculates each block and accumulates the result

**Comment by incrypto32** [comment]
I'll investigate this, from past experience though, almost always its the RPC provider.

@PushinSt do you know which event was supposed to create that `Transaction` entity?

**Comment by incrypto32** [comment]
@PushinSt in the meantime you can use this subgraph on the network. 

https://thegraph.com/explorer/subgraphs/HUZDsRpEVP2AvzDCyzDHtdc64dyDxx8FQjzsmqSg4H3B?view=Playground&chain=arbitrum-one

**Comment by PushinSt** [comment]
> I'll investigate this, from past experience though, almost always its the RPC provider.
> 
> @PushinSt do you know which event was supposed to create that `Transaction` entity?

Yes, I know.
This event (Burn) 
![Screenshot_20230605-093442](https://github.com/graphprotocol/graph-node/assets/36528136/f0d97436-5bc5-4b76-af74-c82a17e1e1c8)

In this transaction 
https://etherscan.io/tx/0xfdeca9f81d0f930e7e107f10b26fd79a26e67b7edf41c683c4d00038e7ccd6f2#eventlog

**Comment by incrypto32** [comment]
Hey @PushinSt, i checked and the timing of that transaction exactly lines up with a previous RPC issue that we faced. So this is not really a graph-node issue, rather a broader problem.

The real solution here is to use the decentralised graph network where this problem is minimized since multiple indexers run multiple graph-node connected to different providers. The hosted service with a single graph-node instance is prone to issues like this caused by centralised points like RPC providers.

Closing this as graph-node is not the problem here
